src/NuGet.Clients/PackageManagement.UI/Actions/UIActionEngine.cs

```diff
@@ -340,7 +340,12 @@ public sealed class UIActionEngine
             return await Task.FromResult(true);
 #else
             var newProjectNames = new List<string>();
-            var msBuildProjects = uiService.Projects.Where(project => project is MSBuildNuGetProject);
+
+            // check if project is packages.config and it's dte project instance can also be converted to VSProject4.
+            // otherwise don't show format selector dialog for this project
+            var msBuildProjects = uiService.Projects.Where(project =>
+                project is MSBuildNuGetProject &&
+                (project as MSBuildNuGetProject).MSBuildNuGetProjectSystem.VSProject4 != null);
```

Comment:
You could just try casting MSBuildNuGetProjectSystem to a VSMSBuildNuGetProjectSystem and then casting EnvDTEProject to VSProject4 yourself.  Exposing VSProject4 just to check if it exists seems unnecessary, and implies that something should be using it externally.

Or you could make this new interface property a bool SupportsPackageReference

Alternatively you oculd